src/ng2-smart-table/components/filter/filter-types/checkbox-filter.component.ts

```diff
@@ -0,0 +1,47 @@
+import { Component, OnInit } from '@angular/core';
+import { Observable } from 'rxjs/Observable';
+
+import { DefaultFilter } from './default-filter';
+import { Column } from '../../../lib/data-set/column';
+
+@Component({
+  selector: 'checkbox-filter',
+  styles: [`
+    a {
+      font-weight: normal;
+    }
+  `],
+  template: `
+    <input type="checkbox" #filterEl [ngClass]="inputClass" class="form-control">
+    <a href="#" *ngIf="filterActive" (click)="resetFilter($event)">{{column.getFilterConfig()?.resetText || 'reset'}}</a>
+  `
+})
+export class CheckboxFilterComponent extends DefaultFilter implements OnInit {
+
+  filterActive: boolean = false;
+
+  constructor() {
+    super();
+  }
+
+  ngOnInit() {
+    this.changesSubscription = Observable.fromEvent(this.filterEl.nativeElement, 'change')
```

Comment:
Thanks for your feedback, I would like to share with you a couple of thoughts:

1. Using Observables we could leverage methods as `.distinctUntilChanged()` and `.debounceTime()` respectively to avoid triggering when the value is unchanged and to trigger the search only after a defined amount of time, this latter thing was previously done by setting and removing a `setTimeout()`. Please let me know if I should remove this _delay_ feature or if I should use your previous `setTimeout()` method.
2. I am not sure that `[(ngModel)]` or `(ngModelChange)` will work since the value of the checkbox does not change, only it's `checked` attribute changes... but I will give it a try!